benches/codecs/newline_bytes.rs

```diff
@@ -0,0 +1,90 @@
+use bytes::BytesMut;
+use criterion::{
+    criterion_group, measurement::WallTime, BatchSize, BenchmarkGroup, BenchmarkId, Criterion,
+    SamplingMode, Throughput,
+};
+use std::{fmt, time::Duration};
+use tokio_util::codec::Decoder;
+use vector::codecs::{self, BytesDeserializer, NewlineDelimitedDecoder};
+
+#[derive(Debug)]
+struct Param {
+    slug: &'static str,
+    input: BytesMut,
+    max_length: Option<usize>,
+}
+
+impl fmt::Display for Param {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.slug)
+    }
+}
+
+fn decoding(c: &mut Criterion) {
+    let mut group: BenchmarkGroup<WallTime> =
+        c.benchmark_group("vector::codecs::decoding::Decoder");
+    group.sampling_mode(SamplingMode::Auto);
+
+    for param in &[
+        Param {
+            slug: "no_max",
+            input: BytesMut::from(include_str!("moby_dick.txt")),
+            max_length: None,
+        },
+        Param {
+            slug: "small_max",
+            input: BytesMut::from(include_str!("moby_dick.txt")),
+            max_length: Some(10),
```

Comment:
Given lines are typically over 10 characters, wouldn't we expect this to mostly just discard bytes?

It might be interesting to see a comparison with max length set to whatever the maximum line length is in input file.

Reply:
It would, yeah. I think what I find surprising is this variant's throughput is not terribly distinct from the 'a' version where the spacing between 'a' would vary. As is, 10 is just kind of a smallish number. What I find suspicious is that the no-max and some-max version are _so_ distinct in throughput, which I think is indicative of the problem we're seeing in #10543. This value was originally 128 with no real throughput difference, but the main area of investigation is why max/no-max differ so much.